doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -668,69 +652,83 @@ Induction
       implicitly added to this table when defined.
       Use the :cmd:`Add` and :cmd:`Remove` commands to update this set manually.
 
-.. tacn:: inversion @ident
-   :name: inversion
+:tacn:`inversion`
+~~~~~~~~~~~~~~~~~
+
+.. tacn:: inversion {| @ident | @natural } using @one_term {? in {+ @ident } }
+          inversion {| @ident | @natural } {? as @or_and_intropattern } {? in {+ @ident } }
+   :name: inversion; _
+
+   .. comment: the other inversion* tactics don't support the using clause,
+      but they should be able to, if desired.  It wouldn't make sense for
+      inversion_sigma.
+      See https://github.com/coq/coq/pull/14179#discussion_r642193096
+
+   .. todo PR: following is clear as mud
 
    Let the type of :n:`@ident` in the local context be :g:`(I t)`, where :g:`I`
-   is a (co)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
+   is a (co-)inductive predicate. Then, ``inversion`` applied to :n:`@ident`
    derives for each possible constructor :g:`c i` of :g:`(I t)`, all the
    necessary conditions that should hold for the instance :g:`(I t)` to be
    proved by :g:`c i`.
 
+   :n:`@ident`
+     The name of the hypothesis to apply.
+
+   :n:`@natural`
+     Equivalent to :n:`intros until @natural; inversion @ident`
+     where :n:`@ident` is the identifier for the last introduced hypothesis.
+
+   :n:`{? in {+ @ident } }`
+     When :n:`{+ @ident}` are identifiers in the local context, this causes
+     generalizing on :n:`{+ @ident}` as the initial step of `inversion`.
+     See :tacn:`generalize`.
+
+   .. todo PR: "generalizing"?  As in :n:`generalize @ident__1; ... generalize @ident__n` ?
+
+   :n:`as @or_and_intropattern`
+      Provides names for the variables introduced in each new subgoal.  The
+      :token:`or_and_intropattern` must have one :n:`{* @intropattern }`
+      for each constructor of the type of :n:`(I t)`, given in the order
+      in which the constructors are defined.
+      If there are not enough names, Coq picks fresh names.
+
+     If an equation splits into several
+     equations (because ``inversion`` applies ``injection`` on the equalities it
+     generates), the corresponding :n:`@intropattern` should be in the form of
+     :n:`@or_and_intropattern` (or the equivalent :n:`{*, ( @simple_intropattern ) }`),
+     with the number of entries equal to the number
+     of subequalities obtained from splitting the original equation.
+
+     Here is an example: the ``inversion … as`` variant of
```

Comment:
From the text before this PR, it rather looks to me like "Here is an example" was left in the middle of the paragraph but didn't actually relate to the following text. So please remove it (or link to the actual example that still appears below).

Reply:
Added a link, but not sure it's in the right place.